File: ISAProject/src/main/java/com/app/offers/OfferService.java
package com.app.offers;

import java.util.List;

import com.app.bidder.Bidder;
import com.app.grocerylist.GroceryList;

public interface OfferService {
	
/*	Iterable<Offer> findByRestaurant(Restaurant restaurant);*/
	List<Offer> findByGroceries(GroceryList grocerylist);
	Offer findOne(Long id);
	Offer save(Offer offer);
	void delete(Long id);
	Offer findByBidder(Bidder bidder);
	boolean acceptOffer(Long id);
	boolean rejectOffer(Long id);
	List<Offer> getPastOffers(Bidder b);
	Offer getPastForGroceryList(GroceryList id);
}

File: ISAProject/src/main/webapp/offer/offerService.js
app.service('offerService', ['$http', function($http){
    
    var url='/offers';
	
	this.findAll = function(){
		return $http.get(url);
	}
    this.findOne = function(id){
		return $http.get(url+"/"+id);
	}    
    this.save = function(offer){
		return $http.post(url, offer);
	}
    this.delete = function(id){
		return $http.delete(url+"/"+id);
	}
    this.getActiveOffer = function(email){
		return $http.get(url+"/getActiveOffer"+"?email="+email);
	}
    this.getOffers = function(id){
		return $http.get(url+"/getOffersForGroceryList/"+id);
	}
    this.getPastOffers = function(email){
		return $http.get(url+"/getPastOffers"+"?email="+email);
	}
    this.acceptOffer = function(id){
		return $http.get(url+"/acceptOffer/"+id);
	}
    this.rejectOffer = function(id){
		return $http.get(url+"/rejectOffer/"+id);
	}
   
}]);